vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt

```diff
@@ -0,0 +1,317 @@
+
+if(AFR_IS_TESTING)
+    set(lpc54018_app_dir "${lpc54018_tests_common_dir}")
+else()
+    set(lpc54018_app_dir "${lpc54018_demos_common_dir}")
+endif()
+
+# -------------------------------------------------------------------------------------------------
+# Amazon FreeRTOS Console metadata
+# -------------------------------------------------------------------------------------------------
+
+afr_set_board_metadata(ID "NXP-LPC54018-IoT-Module")
+afr_set_board_metadata(DISPLAY_NAME "LPC54018 IoT Module")
+afr_set_board_metadata(DESCRIPTION "Development kit for ARM® Cortex®-M4 based LPC54018 MCU")
+afr_set_board_metadata(VENDOR_NAME "NXP")
+afr_set_board_metadata(FAMILY_NAME "LPC5401x_54S0xx")
+afr_set_board_metadata(CODE_SIGNER "null")
+afr_set_board_metadata(DATA_RAM_MEMORY "360KB")
+afr_set_board_metadata(PROGRAM_MEMORY "128MB")
+afr_set_board_metadata(SUPPORTED_IDE "IAREmbeddedWorkbench;MCUXpresso")
+afr_set_board_metadata(IDE_IAREmbeddedWorkbench_NAME "IAR Embedded Workbench")
+afr_set_board_metadata(IDE_IAREmbeddedWorkbench_COMPILER "IAR")
+afr_set_board_metadata(IDE_IAREmbeddedWorkbench_RECOMMENDED "true")
+afr_set_board_metadata(IDE_MCUXpresso_NAME "MCUXpresso")
+afr_set_board_metadata(IDE_MCUXpresso_COMPILER "GCC")
+afr_set_board_metadata(IDE_MCUXpresso_RECOMMENDED "false")
+
+# -------------------------------------------------------------------------------------------------
+# Compiler settings
+# -------------------------------------------------------------------------------------------------
+afr_mcu_port(compiler)
+
+# Defined symbols
+set(defined_symbols
+    __REDLIB__
+    MXL12835F
+    __USE_CMSIS
+    BOARD_DEBUG_UART_TYPE=DEBUG_CONSOLE_DEVICE_TYPE_USBCDC
+    IMG_BAUDRATE=96000000
+    XIP_IMAGE
+    BOARD_USE_VIRTUALCOM
+    USB_STACK_USE_DEDICATED_RAM=1
+    SDK_DEBUGCONSOLE=1
+    CPU_LPC54018JET180=1
+    PRINTF_FLOAT_ENABLE=0
+    USB_STACK_FREERTOS
+    USE_RTOS=1
+    FSL_RTOS_FREE_RTOS
+    A_LITTLE_ENDIAN
+    CR_INTEGER_PRINTF
+    USB_DEVICE_CONFIG_LPCIP3511HS=1
+    __MCUXPRESSO
+    DEBUG
+    SDK_OS_FREE_RTOS
+    CPU_LPC54018JET180_cm4
+    CPU_LPC54018JET180
+)
+
+target_compile_definitions(
+    AFR::compiler::mcu_port
+    INTERFACE $<$<NOT:$<COMPILE_LANGUAGE:ASM>>:${defined_symbols}>
+)
+
+set(common_flags "-mcpu=cortex-m4" "-mfpu=fpv4-sp-d16" "-mfloat-abi=hard" "-mthumb")
+
+set(c_flags "-std=gnu99" "-O0" "-fno-common" "-g" "-Wall" "-c" "-ffunction-sections" "-fdata-sections" "-ffreestanding" "-fno-builtin" "-fomit-frame-pointer")
+
+set(asm_flags "-c" "-x" "assembler-with-cpp")
+
+# Compiler flags
+target_compile_options(
+    AFR::compiler::mcu_port  
+    INTERFACE
+        ${common_flags}
+        $<$<NOT:$<COMPILE_LANGUAGE:ASM>>:${c_flags}>
+        -specs=redlib.specs -MMD -MP
+)
+
+# Assembler flags
+target_compile_options(
+    AFR::compiler::mcu_port        
+    INTERFACE
+        ${common_flags}
+        $<$<NOT:$<COMPILE_LANGUAGE:C>>:${asm_flags}>
```

Comment:
Change this to `$<$<COMPILE_LANGUAGE:ASM>:${asm_flags}>`

Reply:
ok